nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core.Test/MockNuGetPackage.cs

```diff
@@ -0,0 +1,419 @@
+using System.IO.Compression;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.Json.Nodes;
+using System.Text.RegularExpressions;
+using System.Xml;
+using System.Xml.Linq;
+using System.Xml.XPath;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Emit;
+
+namespace NuGetUpdater.Core.Test
+{
+    public record MockNuGetPackage(
+        string Id,
+        string Version,
+        XElement[]? AdditionalMetadata = null,
+        (string? TargetFramework, (string Id, string Version)[] Packages)[]? DependencyGroups = null,
+        (string Path, byte[] Content)[]? Files = null)
+    {
+        private static readonly XNamespace Namespace = "http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd";
+        private static readonly XmlWriterSettings WriterSettings = new()
+        {
+            Encoding = Encoding.UTF8,
+            Indent = true,
+        };
+
+        private XDocument? _nuspec;
+        private Stream? _stream;
+
+        public void WriteToDirectory(string localPackageSourcePath)
```

Comment:
This is the interesting part about the fake NuGet packages; a directory is populated with some specific files (comments inline) that allow the directory to be used as a NuGet package source.